java/configuration/src/main/java/sleeper/configuration/properties/table/TableProperty.java

```diff
@@ -54,145 +57,188 @@ public interface TableProperty extends SleeperProperty {
             .build();
 
     // System defined
-    TableProperty SPLIT_POINTS_KEY = named("sleeper.table.splits.key").build();
-    TableProperty DATA_BUCKET = named("sleeper.table.data.bucket").build();
+    TableProperty SPLIT_POINTS_KEY = named("sleeper.table.splits.key")
+            .propertyGroup(TablePropertyGroup.SPLIT_POINTS).build();
+    TableProperty DATA_BUCKET = named("sleeper.table.data.bucket")
+            .propertyGroup(TablePropertyGroup.CONFIGURATION).build();
     // DynamoDBStateStore properties
-    TableProperty ACTIVE_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.active.table").build();
-    TableProperty READY_FOR_GC_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.gc.table").build();
-    TableProperty PARTITION_TABLENAME = named("sleeper.table.metadata.dynamo.partition.table").build();
+    TableProperty ACTIVE_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.active.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
+    TableProperty READY_FOR_GC_FILEINFO_TABLENAME = named("sleeper.table.metadata.dynamo.gc.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
+    TableProperty PARTITION_TABLENAME = named("sleeper.table.metadata.dynamo.partition.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
     TableProperty DYNAMO_STATE_STORE_POINT_IN_TIME_RECOVERY = named("sleeper.table.metadata.dynamo.pointintimerecovery")
             .defaultProperty(DEFAULT_DYNAMO_POINT_IN_TIME_RECOVERY_ENABLED)
+            .propertyGroup(TablePropertyGroup.METADATA)
             .build();
     TableProperty DYNAMODB_STRONGLY_CONSISTENT_READS = named("sleeper.table.metadata.dynamo.consistent.reads")
             .defaultProperty(DEFAULT_DYNAMO_STRONGLY_CONSISTENT_READS)
             .validationPredicate(s -> s.equalsIgnoreCase("true") || s.equalsIgnoreCase("false"))
+            .propertyGroup(TablePropertyGroup.METADATA)
             .build();
     // S3StateStore properties
-    TableProperty REVISION_TABLENAME = named("sleeper.table.metadata.s3.dynamo.revision.table").build();
+    TableProperty REVISION_TABLENAME = named("sleeper.table.metadata.s3.dynamo.revision.table")
+            .propertyGroup(TablePropertyGroup.METADATA).build();
     TableProperty S3_STATE_STORE_DYNAMO_POINT_IN_TIME_RECOVERY = named("sleeper.table.metadata.s3.dynamo.pointintimerecovery")
             .defaultProperty(DEFAULT_DYNAMO_POINT_IN_TIME_RECOVERY_ENABLED)
+            .propertyGroup(TablePropertyGroup.METADATA)
             .build();
 
+    static List<TableProperty> getAllGrouped() {
+        return sortProperties(getAll());
+    }
+
+    static List<TableProperty> sortProperties(List<TableProperty> unsorted) {
+        List<TableProperty> properties = new ArrayList<>(unsorted);
+        properties.sort(Comparator.comparing(p -> TablePropertyGroup.getAll().indexOf(p.getPropertyGroup())));
+        return properties;
+    }
```

Comment:
This sortProperties method seems to fit better in the TablePropertyGroup class. It could share code with InstancePropertyGroup.sortPropertiesByGroup if we pass in the list of all property groups.